blocks/fileio/test/qa_FileIo.cpp

```diff
@@ -0,0 +1,151 @@
+#include <boost/ut.hpp>
+
+#include <gnuradio-4.0/fileio/BasicFileIo.hpp>
+
+#include <gnuradio-4.0/Scheduler.hpp>
+#include <gnuradio-4.0/testing/NullSources.hpp>
+
+#include <fmt/format.h>
+
+namespace {
+using namespace std::chrono_literals;
+template<typename Scheduler>
+auto createWatchdog(Scheduler& sched, std::chrono::seconds timeOut = 2s, std::chrono::milliseconds pollingPeriod = 40ms) {
+    auto externalInterventionNeeded = std::make_shared<std::atomic_bool>(false);
+
+    std::thread watchdogThread([&sched, externalInterventionNeeded, timeOut, pollingPeriod]() {
+        auto timeout = std::chrono::steady_clock::now() + timeOut;
+        while (std::chrono::steady_clock::now() < timeout) {
+            if (sched.state() == gr::lifecycle::State::STOPPED) {
+                return;
+            }
+            std::this_thread::sleep_for(pollingPeriod);
+        }
+        fmt::println("watchdog kicked in");
+        externalInterventionNeeded->store(true, std::memory_order_relaxed);
+        sched.requestStop();
+        fmt::println("requested scheduler to stop");
+    });
+
+    return std::make_pair(std::move(watchdogThread), externalInterventionNeeded);
+}
+
+template<typename DataType>
+void runTest(const gr::blocks::fileio::Mode mode, const std::shared_ptr<gr::thread_pool::BasicThreadPool>& threadPool) {
+    using namespace boost::ut;
+    using namespace gr::blocks::fileio;
+    using namespace gr::testing;
+    using scheduler = gr::scheduler::Simple<>;
+
+    constexpr gr::Size_t nSamples    = 1024U;
+    const gr::Size_t     maxFileSize = mode == gr::blocks::fileio::Mode::multi ? 256U : 0U;
+    std::string          modeName{magic_enum::enum_name(mode)};
+    std::string          fileName = fmt::format("/tmp/gr4_file_sink_test/TestFileName_{}.bin", modeName);
```

Comment:
Could use [std::temp_directory_path](https://en.cppreference.com/w/cpp/filesystem/temp_directory_path) for portability, but we probably have to do some changes once we try to extend to other platforms, anyway, so it can be done then.

Reply:
Thanks for the info. Didn't know about `std::temp_directory_path`.